Kafka branch

annym opened the pull request:
- Kafka Producer App
- Kafka Consumer App
- Kafka Run Test
- Kafka Test Jupyter Notebook + Results (single node scenario)
- Kafka Doc

Commit:
Kafka Scaling Test

- Included changes to constraint the Kafka-Sub apps to the defined cluster and run broker on a different node from the ones used for subscribers

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Kafka Scaling Test

- Included changes to constraint the Kafka-Sub apps to the defined cluster and run broker on a different node from the ones used for subscribers

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Kafka Scaling Test

- Included changes to constraint the Kafka-Sub apps to the defined cluster and run broker on a different node from the ones used for subscribers

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Kafka Scaling Test

- Included changes to constraint the Kafka-Sub apps to the defined cluster and run broker on a different node from the ones used for subscribers

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

kratos7 commented on build.py:
```diff
@@ -15,7 +15,7 @@
 
 name = "hydra"
-url = 'https://github.com/lake-lerna/hydra'
+url = 'https://github.com/sushilks/hydra'
```
probably you need to pull latest hydra.

tahir24434 commented on docs/INSTALL_kafka_test_results.md:
```diff
@@ -0,0 +1,424 @@
+## Test Goal
+
+The goal of this test is to stress the distributed messaging system **Apache Kafka** with increased client load. The general setup is quite simple. A Kafka Producer sends records at a--maximum--pre-defined transmission rate to the Kafka Server (i.e., cluster) which holds on to these records and hands them out to consumer(s). In this test we assume the following considerations:
+
```
Does it make sense to mention about where every component is running. It would make it little easier to understand.
For now, I am assuming that broker, producer and consumer, all are running on single slave node of cluster.

Commit:
Kafka Max Rate Tests

- Kafka-Related Configuration Parameter Support
- Max Rate & Fixed Test Support for Kafka

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Kafka Max Rate Tests

- Kafka-Related Configuration Parameter Support
- Max Rate & Fixed Test Support for Kafka

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Kafka Max Rate Tests

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Kafka Scaling Test (2)

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Lint Error Fix + README

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

kratos7 commented:
This is based on an old master. Please fetch latest changes from upstream into your local branch. Rebase and recommit.

Commit:
PyKafka Client Support

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

kratos7 commented:
Code looks ok in terms of structure. Please consider adding a unit test for your suite.

Commit:
Pykafka Batch Results in readme.kafka

Signed-off-by: Anny Martinez <annyz@plumgrid.com>

Commit:
Delete Kafka.ipynb

Commit:
Delete .DS_Store

Commit:
Merge branch 'master' into kafka-branch

Commit:
Update readme.kafka.md

kratos7 commented:
Looks good.
